### PR TITLE
Update examples for Zendar API v2

### PR DESCRIPTION
**Pull request by hamlittle**
Changelist

- Replaced make-based build with CMake.
- Removed the 0.1 release debian package. Have not added a new package to the repo.
- Removed the old example code, and added new examples.

Adds example programs for

- Listing modes
- Starting shannon_imaging
- Stopping shannon_imaging
- Receiving just telemetry
- Receiving just data products
- Receiving both data and telemetry

### CODE REVIEW

File: examples/receive_all/main.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#include <zendar/api/api.h>
+
+#include <glog/logging.h>
+#include <gflags/gflags.h>
```
Comment (osamasal) [inline examples/receive_all/main.cc:4]: I see that we're using `cout` everywhere in this file. Is there a reason to include `<glog/logging.h> here?

**Review by osamasal — APPROVED** [review]
Looks good. I was wondering if you can add the build instructions to `README.md` to customers can use this to build the samples. I can use the same instructions to build the samples locally too.

File: examples/receive_all/main.cc (hunk 1 above)
Comment (hamlittle) [reply examples/receive_all/main.cc:4]: not really, I can clean that up

**Comment by hamlittle** [comment]
updated the readme

**Comment by hamlittle** [comment]
This is waiting on a couple PRs for the 0.6 release to be merged, then I can re-test with the release and merge if that goes well.

**Pull request merged**